internal/pkg/agent/application/fleet_server_bootstrap.go

```diff
@@ -100,6 +108,60 @@ func FleetServerComponentModifier(serverCfg *configuration.FleetServerConfig) co
 	}
 }
 
+// injectESHosts will take the hosts list in src and add them to the hosts list in dst if they are not there.
```

Comment:
Open question: Should we keep the original host that is part of `dst` or just insert the hosts from `src` (if `src` is valid)?

Reply:
If the user wants `dst` to be completely removed then we should do it, otherwise you are creating a bug where Fleet server maintains a host that possibly isn't valid anymore with no way to remove it.

Before replacing hosts, we should verify that at least one of them is actually valid to save users from bricking their system with simple typos and such.

This is the approach taken in the policy change handler when changing the agent's Fleet connection.

https://github.com/elastic/elastic-agent/blob/c9bb1643c6817879545c9200e1d3d00c02cd3287/internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go#L192-L198

This is probably difficult to do in a component modifier, which leads me to the next question of why we need to do this in a component modifier?

Why can't this be done by updating Fleet server's output unit to include the new hosts?

Reply:
I'm not super familiar with the agent codebase; is there somewhere else that is better suited to insert the output block?

As I understand, the component modifier  (currently) inserts local config (passed through enrollment flags) into the policy; one of these flags is the hosts list. But it also may include tls/proxy information